robottelo/hosts.py

```diff
@@ -1674,14 +1675,31 @@ def get_caller():
                 if frame.function.startswith('test_'):
                     return frame.function
 
-        return Session(
-            session_name=testname or get_caller(),
-            user=user or settings.server.admin_username,
-            password=password or settings.server.admin_password,
+        test_name = testname or get_caller()
+
+        if getattr(self, 'organization') and getattr(self, 'location') and user is None:
+            login = f'{test_name}_{gen_string("alphanumeric")}'
+            password = gen_string('alphanumeric')
+            new_user = self.api.User(
```

Comment:
Why are we creating a new admin user but not use the default one.

The idea is good but would like to understand the reason. This will also impact many tests and those `tests` switching between admin to non-admin and again back to admin will use default admin creds to operate components.

Though this does not sound bad, but probably test does not demand this scenario always.

Reply:
https://github.com/SatelliteQE/robottelo/blob/67f2a6b67a7c1301c78d09b2281e7b4b6b2afb60/pytest_fixtures/core/ui.py#L9 this getting called when we use `with session:`. The user created and which has the module_org,loc set as default makes the dashboard page is set with org and location default selected

Reply:
@omkarkhatavkar is right, using default admin user could cause us troubles in case of multiple xdist workers or when running multiple tests with same user.
And for `tests switching between admin to non-admin` we don't have such tests where we switch users, but in ui_session contextmanager we've ability to pass user/password in case of non-admin user, so same way if one would like to pass default admin then they can pass too with its nailgun object.

Reply:
`when running multiple tests with same user.` This will never happen because we have devided tests with xDist workers where workers are not processes on one satellite but the different satellites. 

Now all the tests allocated to one satellite xDist worker will run all its tests serially and hence there wont be any race condition. We could have hit this in the past if thats the case.

Reply:
Why I would go with this solution is two things:
1. Testing tests with non-default admin user would be great enhancement but that also means we wont have single test with default admin that most of the customers do use.
2. Auto-selected org and loc so that wont waste our time in tests to select the org/loc explicitly. 

We have to brainstorm on these two things the pros and cons.